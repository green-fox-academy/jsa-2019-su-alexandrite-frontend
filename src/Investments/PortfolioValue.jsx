import React from 'react';
import {
  View,
  Text,
  ImageBackground,
  ActivityIndicator,
} from 'react-native';
import { useSelector } from 'react-redux';
import { FontAwesome } from '@expo/vector-icons';
import styles from './styles';
import cardBackground from '../../assets/img/investment/card.png';

export default function () {
  const {
    isLoading,
    totalValue,
    error,
  } = useSelector((state) => state.investments);

  return (
    <ImageBackground
      source={cardBackground}
      style={styles.valueCard}
      imageStyle={{ borderRadius: 6 }}
    >
      {!error ? (
        <>
          {isLoading && <ActivityIndicator size="large" style={styles.loading} color="#fff" />}
          {!isLoading && !error && (
            <>
              <View style={styles.value}>
                <FontAwesome size={22} name="money" color="#fff" />
                <Text style={styles.valueText}>Value</Text>
              </View>
              <Text style={styles.valueNumber}>{totalValue}</Text>
              <Text style={styles.totalText}>Total Investment</Text>
            </>
          )}
        </>
      ) : <Text>{error.message}</Text>}
    </ImageBackground>
  );
}
