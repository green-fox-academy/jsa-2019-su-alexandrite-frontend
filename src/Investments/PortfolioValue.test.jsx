import React from 'react';
import renderer from 'react-test-renderer';
import { useDispatch, useSelector } from 'react-redux';
import PortfolioValue from './PortfolioValue';

jest.useFakeTimers();

jest.mock('react-redux');
jest.mock('../redux/investment/actionCreator');

const SAMPLE_STATE = {
  isLoading: false,
  totalValue: 123,
  error: null,
};

describe('<PortfolioValue />', () => {
  beforeEach(() => {
    useSelector.mockReturnValue(SAMPLE_STATE);
    useDispatch.mockReturnValue(jest.fn(() => { }));
  });

  afterAll(() => {
    jest.resetModules();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = renderer.create(
      <PortfolioValue />,
    );
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <PortfolioValue />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot when loading', () => {
    useSelector.mockReturnValue({ ...SAMPLE_STATE, isLoading: true });
    const tree = renderer.create(
      <PortfolioValue />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot when error', () => {
    useSelector.mockReturnValue({ ...SAMPLE_STATE, error: { message: 'sample_error' } });
    const tree = renderer.create(
      <PortfolioValue />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
