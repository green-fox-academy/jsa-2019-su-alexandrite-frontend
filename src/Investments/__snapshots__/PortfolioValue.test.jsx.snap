// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioValue /> Should render and match the snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#766ef9",
      "borderRadius": 6,
      "elevation": 6,
      "height": 170,
      "shadowColor": "#766ef9",
      "shadowOffset": Object {
        "height": 3,
      },
      "shadowOpacity": 0.7,
      "shadowRadius": 6,
      "width": 720,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 170,
          "width": 720,
        },
        Object {
          "borderRadius": 6,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginBottom": 6,
        "marginRight": 22,
        "marginTop": 15,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginLeft": 5,
        }
      }
    >
      Value
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 40,
        "fontWeight": "bold",
        "marginLeft": 22,
      }
    }
  >
    123
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "marginBottom": 15,
        "marginHorizontal": 22,
        "marginTop": 20,
      }
    }
  >
    Total Investment
  </Text>
</View>
`;

exports[`<PortfolioValue /> Should render and match the snapshot when error 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#766ef9",
      "borderRadius": 6,
      "elevation": 6,
      "height": 170,
      "shadowColor": "#766ef9",
      "shadowOffset": Object {
        "height": 3,
      },
      "shadowOpacity": 0.7,
      "shadowRadius": 6,
      "width": 720,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 170,
          "width": 720,
        },
        Object {
          "borderRadius": 6,
        },
      ]
    }
  />
  <Text>
    sample_error
  </Text>
</View>
`;

exports[`<PortfolioValue /> Should render and match the snapshot when loading 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#766ef9",
      "borderRadius": 6,
      "elevation": 6,
      "height": 170,
      "shadowColor": "#766ef9",
      "shadowOffset": Object {
        "height": 3,
      },
      "shadowOpacity": 0.7,
      "shadowRadius": 6,
      "width": 720,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 170,
          "width": 720,
        },
        Object {
          "borderRadius": 6,
        },
      ]
    }
  />
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "marginTop": 70,
      }
    }
  />
</View>
`;
